### ISSUE TITLE
Introduce clang-formatting

### ISSUE THREAD

**Issue by carme-hp** [opendihu/opendihu#24]
As discussed, we should have some kind of formatting in OpenDiHu. I guess the easiest option is to use clang-format.
I made a first attempt in the [clang-format branch](https://github.com/opendihu/opendihu/tree/clang-format).

> [!NOTE]  
> To use the formatting you first need to install clang-format, e.g., `sudo apt install clang-format`
> To format the files run `. tools/formatting/format-all` in the parent directory.

> [!WARNING]  
> The script `format-all` does not actually formats everything. It formats the core, examples, testing and scripts folders. For the remaining folders I found error messages that I have not yet addressed:
> In case of docs and tools: _error: cannot use -i when reading from stdin_
> In case of dependencies: _Configuration file(s) do(es) not support Objective-C: /home/homspoce/Software/opendihu/.clang-format_

**Comment by Conni2461** [comment]
:+1: would really appreciate these changes, some questions i just wanna leave here:
- do you wanna copy the configuration file found in precice repo? https://github.com/precice/precice/blob/develop/.clang-format
- do you wanna introduce a ci pipeline to verify correct formatting 
  - precice does this using pre commit hooks: https://github.com/precice/precice/blob/develop/.github/workflows/code-style.yml and https://github.com/precice/precice/blob/develop/.pre-commit-config.yaml
  - i usually do this with a simple shell script https://github.com/nvim-telescope/telescope-fzf-native.nvim/blob/main/.github/workflows/lint.yml#L21-L31 and https://github.com/nvim-telescope/telescope-fzf-native.nvim/blob/main/Makefile#L32-L34

**Comment by ghoshgm** [comment]
@Conni2461 Verification of code formatting would be good idea. There are some caveats with formatting open-dihu code, after fixing that we can focus on verification.

As for code style, I also thought about copying over preCICE format but I guess we can decide if we want add our touch to it. That can discussed on Monday.

**Comment by carme-hp** [comment]
I tried using the llvm style guide  (.clang-format is automatically generated for default styles) with the idea to  modify if necessary for OpenDiHu. 

It's still a WIP.

**Comment by carme-hp** [comment]
The code is now successfully formatted using the llvm style guide. 
- There was the issue about the miss-ordering of the includes. My first idea was to write priorities for the affected files but i didn't manage it for it to work, so instead I just did `SortIncludes:    false`.
- For checking clang-format I used @Conni2461 's approach:)
- Refer to the [readme.md](https://github.com/opendihu/opendihu/blob/clang-format/tools/formatting/readme.md) files to use/check clang-format
